test/src/module/common/walFilterGPDB6Test.c

```diff
@@ -186,6 +186,40 @@ calcBodySize(uint32 pageCount, uint32 leftOnLastPage, XLogRecPtr recPtr, uint32
     return totalSize;
 }
 
+static void
+fillWalWithRecords(Buffer *wal, uint32 targetOffset, uint32 segSize)
+{
+    if (targetOffset == SizeOfXLogLongPHD)
+    {
+        return;
+    }
+
+    // We stop when we get close to the target offset by the page size.
+    while (targetOffset - bufUsed(wal) > DEFAULT_GDPB_XLOG_PAGE_SIZE)
```

Comment:
This substraction of unsigned ints looks dangerous. What if bufUsed(wal) > targetOffset?

The same note about 212 line. Maybe add additional checking.